tensorflow_serving/batching/batching_util.h

```diff
@@ -0,0 +1,206 @@
+/* Copyright 2017 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_SERVING_BATCHING_BATCHING_UTIL_H_
+#define TENSORFLOW_SERVING_BATCHING_BATCHING_UTIL_H_
+
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "tensorflow/contrib/batching/batch_scheduler.h"
+#include "tensorflow/core/framework/tensor.h"
+#include "tensorflow/core/lib/core/status.h"
+
+
+
+namespace tensorflow {
+namespace serving {
+
+// Constructs array of paddings, where:
+// paddings[i].first - number of objects to add before elements in dimension i
+// of given tensor,
+// paddings[i].second - number of objects to add after elements in dimension i.
+// This padding signature is used when performing internal padding with Eigen.
+//
+// When building paddings it is assumed that tensor needs to be padded
+// after each dimension, so its shape matches max_dim_sizes,
+// except zeroth dimension which is left as is.
+template<int num_dims>
+std::array<std::pair<int32, int32>, num_dims> CreatePadding(Tensor tensor,
```

Comment:
Well, I pass the value returned by this function to pad method of Eigen::TensorMap. When I try to pass std::vector instead of std::array it gives me compilation error saying "template instantiation depth exceeds maximum of 900" on [this line](https://github.com/RLovelett/eigen/blob/master/unsupported/Eigen/CXX11/src/util/CXX11Meta.h#L362).  There appears to be some recursive template array unfolding. It seems that Eigen needs to know size of array at compile time.